packages/airnode-deployer/src/infrastructure/aws.test.ts

```diff
@@ -0,0 +1,315 @@
+import fs from 'fs';
+import {
+  getAirnodeBucket,
+  createAirnodeBucket,
+  getBucketDirectoryStructure,
+  storeFileToBucket,
+  getFileFromBucket,
+  copyFileInBucket,
+  deleteBucketDirectory,
+  deleteBucket,
+} from './aws';
+import { mockBucketDirectoryStructure, mockBucketDirectoryStructureList } from '../../test/fixtures';
+import { Directory } from '../utils/infrastructure';
+
+const mockPromise = (fn: Function) => () => ({ promise: fn });
+
+const mockS3 = {
+  listBuckets: jest.fn(),
+  createBucket: jest.fn(),
+  putBucketEncryption: jest.fn(),
+  putPublicAccessBlock: jest.fn(),
+  listObjectsV2: jest.fn(),
+  putObject: jest.fn(),
+  getObject: jest.fn(),
+  copyObject: jest.fn(),
+  deleteObjects: jest.fn(),
+  deleteBucket: jest.fn(),
+};
+
+jest.mock('aws-sdk', () => ({
+  config: {
+    update: jest.fn().mockImplementation(() => {}),
+  },
+  S3: jest.fn(() => mockS3),
+}));
+
+jest.mock('../utils/infrastructure', () => ({
+  ...jest.requireActual('../utils/infrastructure'),
+  generateBucketName: jest.fn(),
+}));
+
+const awsConfigUpdateSpy = jest.requireMock('aws-sdk').config.update;
+const awsListBucketsSpy: jest.SpyInstance = jest.requireMock('aws-sdk').S3().listBuckets;
```

Comment:
Nit: You could just do `const abc = mockS3.xyz` or just use `mock.xyz` in the tests as well. (But I also like the explicitness). Leaving up to you.